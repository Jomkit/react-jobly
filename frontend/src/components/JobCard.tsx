import { Button, Card, CardBody, CardTitle, Col, Row } from 'reactstrap'
import { jobsInterface } from '../types'

const JobCard = ({ title, salary, equity, companyName }: jobsInterface) => {
  return (
    <li>
        <Card className='p-1 pt-3'>
            <CardTitle tag="h5" className='text-start mx-3'>
                {title}
            </CardTitle>
            <CardBody>
                <Row>
                    <Col sm="8" className='text-start'>
                        <p>{ companyName }</p>
                        <p>{ salary }</p>
                        <p>{ equity }</p>
                    </Col>
                    <Col sm="4" className='text-end mt-auto pb-2'>
                        <Button className='bg-danger'>Apply</Button>
                    </Col>
                </Row>
            </CardBody>
            
        </Card>
    </li>
  )
}

export default JobCard