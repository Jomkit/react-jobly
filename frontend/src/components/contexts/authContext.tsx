import { createContext } from 'react';
import { authInterface } from '../../types';

export const authContext = createContext<authInterface>({} as authInterface);