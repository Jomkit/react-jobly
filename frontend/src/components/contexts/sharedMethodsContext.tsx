import { createContext } from 'react';
import { sharedMethodsInterface } from '../../types';

export const sharedMethodsContext = createContext<sharedMethodsInterface>({} as sharedMethodsInterface);