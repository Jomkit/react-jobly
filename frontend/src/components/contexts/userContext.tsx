import { createContext } from "react";
import { userInterface } from "../../types";

export const userContext = createContext<userInterface | null>(null);