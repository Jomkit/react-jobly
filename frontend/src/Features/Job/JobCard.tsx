import { Button, Card, CardBody, CardTitle, Col, Row } from 'reactstrap'
import { jobsInterface, userInterface } from '../../types'
import { useContext, useState } from 'react'
import { sharedMethodsContext } from '../../components/contexts/sharedMethodsContext'
import { userContext } from '../../components/contexts/userContext'

const JobCard = ({ id, title, salary, equity, companyName }: jobsInterface) => {
    const currUser: userInterface | null = useContext(userContext);
    const { applyToJob } = useContext(sharedMethodsContext);
    const [ isDisabled, setIsDisabled ] = useState(false);
    
    const handleClick = (evt: any) => {
        if(applyToJob){
            applyToJob(evt.target.value);
            setIsDisabled(true);
        }
    }
    
  if(currUser) return (
    <li>
        <Card className='p-1 pt-3 m-5 border border-3 border-black'>
            <CardTitle tag="h5" className='text-start mx-3'>
                {title}
            </CardTitle>
            <CardBody>
                <Row>
                    <Col sm="8" className='text-start'>
                        <p>{ companyName }</p>
                        <p>Salary: { salary }</p>
                        <p>Equity: { equity === null ? "n/a" : (+equity).toFixed(2) }</p>
                    </Col>
                    <Col sm="4" className='text-end mt-auto pb-2'>
                    { currUser.applications && currUser.applications.includes(id) ? 
                        <Button className='bg-danger' onClick={handleClick} disabled>Apply</Button>
                        :
                        <Button className='bg-danger' value={id} onClick={handleClick} disabled={isDisabled}>Apply</Button>
                    }
                    </Col>
                </Row>
            </CardBody>
            
        </Card>
    </li>
  )
}

export default JobCard