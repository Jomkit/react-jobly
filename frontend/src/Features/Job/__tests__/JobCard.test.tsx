import { render } from "@testing-library/react"
import { TESTJOBS, TESTUSER } from "../../../__tests__/_testCommon";
import JobCard from "../JobCard";
import { MemoryRouter } from "react-router-dom";
import { userContext } from "../../../components/contexts/userContext";

const { id, title, salary, equity, companyName } = TESTJOBS[0];

const TestJobCard = () => {
    return (
        <MemoryRouter>
            < userContext.Provider value={TESTUSER}>
                <JobCard key={id} id={id} title={title} salary={salary} equity={equity} companyName={companyName } />
            </userContext.Provider>
        </MemoryRouter>
    )
}

it('should render without crashing', () => { 
    render(<TestJobCard />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<TestJobCard />);
    expect(asFragment()).toMatchSnapshot();
 })
