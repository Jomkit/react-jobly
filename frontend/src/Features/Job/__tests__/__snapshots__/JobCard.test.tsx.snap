// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <li>
    <div
      class="p-1 pt-3 m-5 border border-3 border-black card"
    >
      <h5
        class="text-start mx-3 card-title"
      >
        test job 1
      </h5>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="text-start col-sm-8"
          >
            <p>
              test company 1
            </p>
            <p>
              Salary: 1000
            </p>
            <p>
              Equity: 0.10
            </p>
          </div>
          <div
            class="text-end mt-auto pb-2 col-sm-4"
          >
            <button
              class="bg-danger btn btn-secondary"
              type="button"
              value="1"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
